plugin/pulpcore/plugin/download/http.py

```diff
@@ -117,7 +117,8 @@ def __init__(self, url, session=None, auth=None, proxy=None, proxy_auth=None,
             self.session = session
             self._close_session_on_finalize = False
         else:
-            self.session = aiohttp.ClientSession()
+            timeout = aiohttp.ClientTimeout()
```

Comment:
I am wondering whether this leads to the opposite extreme case that a download connection may hang forever if the peer disappears? (established TCP connections can live forever if no data is sent and keep-alives are disabled)

IMHO, we should set `sock_read` and `sock_connect` if there is no total timeout. Something needs to kill a dead connection or a connection attempt to a "black hole".

Reply:
@gmbnomis I'm so glad you brought. I agree when keepalive is disabled at the TCP level or in a buggy proxy we would fall prey to [these issues](http://www.tldp.org/HOWTO/TCP-Keepalive-HOWTO/overview.html).

What about `sock_connect` and `sock_read` both set to 5 minutes? If a server can't respond in 5 minutes or new data can't be read in 5 minutes I think the remote side is dead.

Reply:
@bmbouter Makes sense. I would have chosen shorter values, but this is only gut feeling. There is no "one value fits all situations" here anyway. The important part is that downloads do not hang forever; tuning can happen later if necessary.

Reply:
@gmbnomis I had similar feelings. I really want to avoid false_positives; perhaps too much. +1 to more tuning later. Thank you so much for the input. It's much appreciated.